documentation/cluster-controller/cluster-controller.adoc

```diff
@@ -0,0 +1,296 @@
+cluster configuration. Such ConfigMaps need to have a specific label which is, by default, `strimzi.io/kind=cluster`
+(as described <<Format of the cluster ConfigMap,later>>) that can be changed through a corresponding environment variable.
+
+When a new ConfigMap is created in the Kubernetes/OpenShift cluster, the controller gets the cluster configuration from
+its `data` section and starts creating a new Kafka or Kafka Connect cluster by creating the necessary Kubernetes/OpenShift
+resources, such as StatefulSets, ConfigMaps, Services etc.
+
+Every time the ConfigMap is updated by the user with some changes in the `data` section, the controller performs corresponding
+updates on the Kubernetes/OpenShift resources which make up the Kafka or Kafka Connect cluster. Resources are either patched
+or deleted and then re-created in order to make the Kafka or Kafka Connect cluster reflect the state of the cluster ConfigMap.
+This might cause a rolling update which might lead to service disruption.
+
+Finally, if the ConfigMap is deleted, the controller starts to un-deploy the cluster deleting all the related Kubernets/OpenShift
+resources.
+
+== Reconciliation
+
+When the controller is running it reacts to all notifications about the cluster ConfigMaps received from the Kubernetes/OpenShift cluster.
+However if the controller is not running, or if a notification is not received for any reason it could lose notifications
+about the cluster ConfigMaps.
+
+For different reasons, it could lose notifications about the cluster ConfigMaps; without any action there will be an
+inconsistency between the cluster ConfigMaps and the state of the running Kubernetes/OpenShift cluster.
+
+In order to handle failovers properly, a periodic reconciliation process is executed by the cluster controller so
+that it can compare the state of the ConfigMaps with the current cluster deployment in order to have
+a consistent state across all of them.
+
+== Format of the cluster ConfigMap
+
+By default, the controller watches for ConfigMaps having the label `strimzi.io/kind=cluster` in order to find and get
+configuration for a Kafka or Kafka Connect cluster to deploy. The label is configurable through the `STRIMZI_CONFIGMAP_LABELS` 
+environment variable.
+
+In order to distinguish which "type" of cluster to deploy, Kafka or Kafka Connect, the controller checks the
+`strimzi.io/type` label which can have one of the the following values :
+
+* `kafka`: the ConfigMap provides configuration for a Kafka cluster (with Zookeeper ensemble) deployment
+* `kafka-connect`: the ConfigMap provides configuration for a Kafka Connect cluster deployment
+* `kafka-connect-s2i`: the ConfigMap provides configuration for a Kafka Connect cluster deployment using Build and Source2Image
+features (works only with OpenShift)
+
+The `data` section of such ConfigMaps contains different keys depending on the "type" of deployment as described in the 
+following sections.
+
+=== Kafka
```

Comment:
I think it would be great if all these sections contained also a list of resources we create and their names (i.e. tell the users that the stateful set will be named _[cluster-name]-kafka_, so they should try to avoid having cluster named _kafka_). This will make it also easier for them to not delete anything and understand the RBAC rights.